### PR TITLE
Moved Pagination inside Listing React component

### PR DESCRIPTION
**Pull request by noisecapella**
Prerequisite for #736 

This just moves the Pagination inside the Listing component so we can more easily deal with the whole area at once

### CODE REVIEW

**Comment by jamiefolsom** [comment]
Hi @noisecapella -- reviewing this one now.

File: ui/static/ui/js/listing_resources.jsx [hunk 1]
```diff
@@ -430,6 +432,7 @@ define('listing_resources', ['react', 'jquery', 'lodash', 'utils'],
     var ListingPage = React.createClass({
       render: function() {
         return <div>
```
Comment (jamiefolsom) [inline ui/static/ui/js/listing_resources.jsx:434]: One thought (not necessary to implement with this PR, from my pt of view). Would it make sense to always add in an id or a class to divs within react components (and possibly matching close div comments), so we can hook into them more easily from CSS, and identify their functions when viewed in the browser? If we have a lot of vanilla `<div>`s sprinkled around, it could get harder over time to keep things straight. Maybe react gives us what we need? React ids tend to be inscrutable to the naked eye. Thoughts?

Comment (noisecapella) [inline ui/static/ui/js/listing_resources.jsx:434]: We should avoid `id` whenever possible since they may conflict and we would prevent easy reuse of components. CSS classes are fine but I prefer to add those when we have something to style, or maybe as identification for tests.

**Comment by jamiefolsom** [comment]
- [x] functions properly
- [x] js tests pass
- [x] no console errors
- [x] code looks good
- [x] Question: is this rebased against master? git diff origin/master is a big diff, whereas the PR isn't many lines.

**Comment by jamiefolsom** [comment]
I am getting a 500 error. Here's the console log output.

```
web_1     | [07/Oct/2015 14:55:13] "GET / HTTP/1.1" 200 4748
web_1     | [07/Oct/2015 14:55:13] "GET /static/ui/css/mit-lore.css HTTP/1.1" 304 0
web_1     | [07/Oct/2015 14:55:13] "GET /static/ui/js/require_config.js HTTP/1.1" 304 0
web_1     | [07/Oct/2015 14:55:13] "GET /static/CACHE/js/require.372f9f3fb615.js HTTP/1.1" 200 83559
web_1     | [2015-10-07 14:55:13] WARNING 27 [django.request] base.py:170 - [73f8202f21e0] - Not Found: /favicon.ico
web_1     | [07/Oct/2015 14:55:13] "GET /favicon.ico HTTP/1.1" 404 3572
web_1     | [07/Oct/2015 14:55:20] "GET /repositories/test/ HTTP/1.1" 200 10300
web_1     | [07/Oct/2015 14:55:20] "GET /static/CACHE/js/listing_resources.fbf4889297e5.js HTTP/1.1" 200 18998
web_1     | [07/Oct/2015 14:55:20] "GET /static/CACHE/js/require.372f9f3fb615.js HTTP/1.1" 200 83559
web_1     | [07/Oct/2015 14:55:20] "GET /static/CACHE/js/lr_exports.038ecbc9fd74.js HTTP/1.1" 200 7870
web_1     | [07/Oct/2015 14:55:20] "GET /static/CACHE/js/manage_taxonomies.ac519fffdf84.js HTTP/1.1" 200 29092
web_1     | [07/Oct/2015 14:55:20] "GET /static/CACHE/js/static_assets.e86b226f8bab.js HTTP/1.1" 200 2366
web_1     | [07/Oct/2015 14:55:20] "GET /static/CACHE/js/learning_resources.554e27260904.js HTTP/1.1" 200 15535
web_1     | [07/Oct/2015 14:55:20] "GET /static/CACHE/js/xml_panel.b6a12c926131.js HTTP/1.1" 200 2096
web_1     | [07/Oct/2015 14:55:20] "GET /static/CACHE/js/pagination.f8e634aa8c38.js HTTP/1.1" 200 3144
web_1     | [07/Oct/2015 14:55:20] "GET /static/CACHE/js/utils.1ae565e1dd27.js HTTP/1.1" 200 11814
web_1     | [07/Oct/2015 14:55:20] "GET /static/ui/js/listing.js HTTP/1.1" 200 17854
web_1     | [07/Oct/2015 14:55:20] "GET /static/ui/js/csrf.js HTTP/1.1" 304 0
web_1     | [07/Oct/2015 14:55:20] "GET /static/bower/uri.js/src/URI.js HTTP/1.1" 304 0
web_1     | [07/Oct/2015 14:55:20] "GET /static/bower/history.js/scripts/compressed/history.adapter.jquery.js HTTP/1.1" 304 0
web_1     | [07/Oct/2015 14:55:21] "GET /static/bower/uri.js/src/SecondLevelDomains.js HTTP/1.1" 304 0
web_1     | [07/Oct/2015 14:55:21] "GET /static/bower/uri.js/src/punycode.js HTTP/1.1" 304 0
web_1     | [07/Oct/2015 14:55:21] "GET /static/bower/history.js/scripts/compressed/history.js HTTP/1.1" 304 0
web_1     | [07/Oct/2015 14:55:21] "GET /static/bower/uri.js/src/IPv6.js HTTP/1.1" 304 0
web_1     | [07/Oct/2015 14:55:21] "GET /static/bower/react-infinite/dist/react-infinite.min.js HTTP/1.1" 304 0
web_1     | [2015-10-07 14:55:21] ERROR 27 [django.request] base.py:256 - [73f8202f21e0] - Internal Server Error: /api/v1/repositories/test/search/
web_1     | Traceback (most recent call last):
web_1     |   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 132, in get_response
web_1     |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
web_1     |   File "/usr/local/lib/python2.7/dist-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
web_1     |     return view_func(*args, **kwargs)
web_1     |   File "/usr/local/lib/python2.7/dist-packages/rest_framework/viewsets.py", line 87, in view
web_1     |     return self.dispatch(request, *args, **kwargs)
web_1     |   File "/usr/local/lib/python2.7/dist-packages/rest_framework/views.py", line 466, in dispatch
web_1     |     response = self.handle_exception(exc)
web_1     |   File "/usr/local/lib/python2.7/dist-packages/rest_framework/views.py", line 463, in dispatch
web_1     |     response = handler(request, *args, **kwargs)
web_1     |   File "/usr/local/lib/python2.7/dist-packages/statsd/client.py", line 30, in _wrapped
web_1     |     return_value = f(*args, **kwargs)
web_1     |   File "/src/rest/views.py", line 921, in list
web_1     |     response = self.get_paginated_response(serializer.data)
web_1     |   File "/usr/local/lib/python2.7/dist-packages/rest_framework/serializers.py", line 658, in data
web_1     |     ret = super(ListSerializer, self).data
web_1     |   File "/usr/local/lib/python2.7/dist-packages/rest_framework/serializers.py", line 223, in data
web_1     |     self._data = self.to_representation(self.instance)
web_1     |   File "/usr/local/lib/python2.7/dist-packages/rest_framework/serializers.py", line 598, in to_representation
web_1     |     self.child.to_representation(item) for item in iterable
web_1     |   File "/usr/local/lib/python2.7/dist-packages/rest_framework/serializers.py", line 447, in to_representation
web_1     |     attribute = field.get_attribute(instance)
web_1     |   File "/usr/local/lib/python2.7/dist-packages/rest_framework/fields.py", line 401, in get_attribute
web_1     |     raise type(exc)(msg)
web_1     | AttributeError: Got AttributeError when attempting to get a value for field `description_path` on serializer `RepositorySearchSerializer`.
web_1     | The serializer field might be named incorrectly and not match any attribute or key on the `Result` instance.
web_1     | Original exception text was: 'Result' object has no attribute 'description_path'.
web_1     | [07/Oct/2015 14:55:21] "GET /api/v1/repositories/test/search/ HTTP/1.1" 500 24495

```

**Comment by noisecapella** [comment]
Can you run ./manage.py recreate_index and try again?

**Comment by jamiefolsom** [comment]
Sure thing.

**Comment by noisecapella** [comment]
I just rebased on master by the way

**Comment by jamiefolsom** [comment]
OK, will re-pull. Maybe that will fix this issue too: `recreate_index` fails out with the following console output (note that upon restarting, docker reports recreating a bunch of things):

```
11:00 $ docker-compose run web python manage.py recreate_index
Starting lore_stats_1...
Starting lore_elastic_1...
Starting lore_redis_1...
Starting lore_db_1...
Traceback (most recent call last):
  File "manage.py", line 16, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 338, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 330, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 393, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 444, in execute
    output = self.handle(*args, **options)
  File "/src/search/management/commands/recreate_index.py", line 20, in handle
    clear_index()
  File "/src/search/utils.py", line 265, in clear_index
    conn = get_conn()
  File "/src/search/utils.py", line 51, in get_conn
    if not _CONN.indices.exists(INDEX_NAME):
  File "/usr/local/lib/python2.7/dist-packages/elasticsearch/client/utils.py", line 69, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/elasticsearch/client/indices.py", line 216, in exists
    self.transport.perform_request('HEAD', _make_path(index), params=params)
  File "/usr/local/lib/python2.7/dist-packages/elasticsearch/transport.py", line 307, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "/usr/local/lib/python2.7/dist-packages/elasticsearch/connection/http_urllib3.py", line 85, in perform_request
    raise ConnectionError('N/A', str(e), e)
elasticsearch.exceptions.ConnectionError: ConnectionError(<urllib3.connection.HTTPConnection object at 0x7f94aeb529d0>: Failed to establish a new connection: [Errno 111] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x7f94aeb529d0>: Failed to establish a new connection: [Errno 111] Connection refused)
✘-1 ~/repos/lore [refactor/gs/pagination_in_listing ↓·5↑·1|⚑ 12]
11:01 $ docker-compose up
Recreating lore_stats_1...
Recreating lore_elastic_1...
Recreating lore_redis_1...
Recreating lore_db_1...
Recreating lore_web_1...
Recreating lore_celery_1...
```

**Comment by jamiefolsom** [comment]
I now get this error on recreate_index:

```
11:07 $ docker-compose run web python manage.py recreate_index
No handlers could be found for logger "elasticsearch.trace"
```

And I get the same 500 error on loading the listing page. 

Could be I need to run docker-compose build. Let me try that.

**Comment by jamiefolsom** [comment]
OK, docker-compose build appears to have resolved those problems. Looking at functionality now.

File: ui/templates/data.html [hunk 2]
```diff
@@ -11,13 +11,6 @@
 <div id="data">
```
Comment (jamiefolsom) [inline ui/templates/data.html:11]: Nit: both for the id of this div, and for the name of the include file, "data" is pretty generic. Could this be called "listing-search-results" or similar? Not a blocker; I'm just casting about for suggestions on an otherwise clean, tidy PR.

**Comment by jamiefolsom** [comment]
Couple of stylistic nits, otherwise :+1: from me!

File: ui/templates/data.html (hunk 2 above)
Comment (noisecapella) [inline ui/templates/data.html:11]: I agree but this is outside the scope of the PR. By the way this is where the datatable goes when you visit /repositories/repo/data/

**Comment by jamiefolsom** [comment]
Appears the PR build failed on Travis, due to an elastic search connection timeout. Might just rerun that, @noisecapella if you think it's unrelated.

File: ui/static/ui/js/listing_resources.jsx (hunk 1 above)
Comment (jamiefolsom) [inline ui/static/ui/js/listing_resources.jsx:434]: What would an id conflict with? If classes are better, that's fine -- identification is my main concern.

Comment (noisecapella) [inline ui/static/ui/js/listing_resources.jsx:434]: If we had two `ListingPage` components the id would conflict with each other. If we need to identify a particular DOM node within the React component then we can use `ref` instead.

**Comment by jamiefolsom** [comment]
Good to merge @noisecapella!

File: ui/static/ui/js/listing_resources.jsx (hunk 1 above)
Comment (jamiefolsom) [inline ui/static/ui/js/listing_resources.jsx:434]: Of course, makes sense -- forgot we might use several of the same component on a page. Carry on!

File: ui/templates/data.html (hunk 2 above)
Comment (jamiefolsom) [inline ui/templates/data.html:11]: Makes sense.
